packages/slate-editor/src/modules/editor-v4-floating-add-menu/lib/shouldShowMenuButton.ts

```diff
@@ -12,7 +12,11 @@ export function shouldShowMenuButton(editor: Editor): boolean {
 
     const [currentNode] = EditorCommands.getCurrentNodeEntry(editor) || [];
 
-    if (!isParagraphNode(currentNode) || EditorCommands.hasVoidElements(editor, currentNode)) {
+    if (
+        !isParagraphNode(currentNode) ||
+        isHeadingNode(currentNode) ||
```

Comment:
I don't understand this condition. Isn't `isHeadingNode` effectively a noop? Because `!isParagraphNode` already includes a case when the node is a heading node.

Reply:
I think it should be changed to `!(isParagraphNode(currentNode) || isHeadingNode(currentNode) || hasVoidNodes(...))`.

Or even better split it into two conditions to make things clear:

```ts
if (!isParagraphNode(currentNode) && !isHeadingNode(currentNode)) {
   return false;
}
if (hasVoidNodes(...)) {
   return false.
}
```